### PR TITLE
Call unminimization ealier

### PR DESCRIPTION
**Pull request by cemiarni**

### CODE REVIEW

File: bldr/data/hooks/pre-init [hunk 1]
```diff
@@ -1 +1,26 @@
...
+        echo
+        cat <<"END"
+Performing an image unminimization, because we experienced that more and more
+packages have tests depending on data which is not present on a minimized
+image.
+
+If this unminimization bothers you, please remove the
+'perform_unminimization' function from the 'pre-install-deps' hook.
+END
+        echo
```
Comment (wferi) [inline bldr/data/hooks/pre-init:16]: A single echo would do instead of echo+cat+here doc+echo here.

File: bldr/data/hooks/pre-init [hunk 2]
```diff
@@ -1 +1,26 @@
...
+
+    if [[ "${unminimize_cmd}" != "" ]]; then
+        echo
+        cat <<"END"
+Performing an image unminimization, because we experienced that more and more
+packages have tests depending on data which is not present on a minimized
+image.
+
+If this unminimization bothers you, please remove the
+'perform_unminimization' function from the 'pre-install-deps' hook.
```
Comment (wferi) [inline bldr/data/hooks/pre-init]: from the 'pre-init' hook

File: bldr/data/hooks/pre-init [hunk 3]
```diff
@@ -1 +1,26 @@
...
+END
+        echo
+
+        set -e
+        yes | "${unminimize_cmd}"
+        set +e
```
Comment (wferi) [inline bldr/data/hooks/pre-init]: What about a more explicit

    yes | "${unminimize_cmd}" || exit

or similar?

**Review by wferi — CHANGES_REQUESTED** [review]

**Review by wferi — COMMENTED** [review]
Commit message recommendation:

bldr/data/Dockerfile.tpl: move fake local_apt repo creation earlier

The `apt update` command did not work in the `pre-init` hook because
the `local_apt` repository did not exist yet. Only the
`init-docker-image` script created an empty repository (which is replaced
by a bind mount when the container is running) to make `apt update` work
at image build time. But `init-docker-image` is called after `pre-init`,
so we have to move the empty repo creation before calling the `pre-init` hook in
`Dockerfile.tpl`.

**Review by wferi — CHANGES_REQUESTED** [review]
Please give an example of something going wrong due to late unminimization. I agree the sooner we do it the better, but it isn't obvious why that isn't a pure efficiency concern. What does it not reinstall properly if done late?

Some typos are already fixed below:

> bldr/data/hooks/pre-init: move perform_unminimization to pre-init
>
> The unminimization is supposed to be performed as early as possible,
because it does not reinstall all packages.
The `unminimize` command tells us the following: 'This script restores
content and packages that are found on a default Ubuntu server
system...'
>
> Add some extra double quotes to `perform_uniminization` to make it more robust.

File: bldr/data/hooks/pre-init (hunk 2 above)
Comment (kocsob) [reply bldr/data/hooks/pre-init]: +1

File: bldr/data/hooks/pre-init (hunk 1 above)
Comment (cemiarni) [reply bldr/data/hooks/pre-init:16]: Yes it would, but I would not like to change it.

File: bldr/data/hooks/pre-init [hunk 4]
```diff
@@ -1 +1,24 @@
...
+Performing an image unminimization, because we experienced that more and more
+packages have tests depending on data which is not present on a minimized
+image.
+
+If this unminimization bothers you, please remove the
+'perform_unminimization' function from the 'pre-init' hook.
+END
+        echo
+
+        yes | "${unminimize_cmd}" | exit 1
```
Comment (wferi) [inline bldr/data/hooks/pre-init]: | != ||

**Review by wferi — CHANGES_REQUESTED** [review]

**Pull request merged**